### DISCUSSION TITLE
Python: keygen_licensing_tools

### DISCUSSION THREAD

**Discussion by nschloe** [keygen-sh/community#11] [Show and tell]
I just released a first version of [keygen_licensing_tools](https://github.com/cryptomakerbr/keygen_licensing_tools), a Python package for easy interaction with the Keygen API. Can do online license validation and (secure) cached license validation for now.

These are my first few days with Keygen. Observations so far:

* I love how progressive and lightweight Keygen is. Definitely has that startup-with-a-smart-leader vibe. I'm feeling very comfortable using it. :couch_and_lamp: 
* Opinion: I don't like the name _Keygen_, because it does so much more than key generation. Find it misleading.
* Documentation is great. :rocket:

**Comment by ezekg** [comment]
This is great! Thanks for building this. I'm sure it'll be very useful for other Python developers. 🤙

Would you be open to me adding a link to it here, https://keygen.sh/docs/api/#client-libraries?

**Reply by nschloe** [reply]
Sure thing.

**Reply by nschloe** [reply]
Btw I'd love to do proper testing on the Python package, but for that I'd need more demo data, e.g., the verify key of the `demo` account, the public RSA key, and some offline keys.

**Reply by ezekg** [reply]
We don’t provide access to the demo account but feel free to create your own!

**Comment by antonbabenko** [comment]
@nschloe Have you moved/deleted your project? I'd like to check it out and probably replace my solution.

**Reply by antonbabenko** [reply]
I found it here - https://github.com/cryptomakerbr/keygen_licensing_tools :)

**Reply by ezekg** [reply]
Glad you found it!
